test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/AsyncSignerOverrideTest.java

```diff
@@ -58,4 +58,10 @@ public void test_signerOverriddenForStreamingInput_takesPrecedence() {
 
         verify(mockSigner).sign(any(SdkHttpFullRequest.class), any(ExecutionAttributes.class));
     }
+
+    // TODO(sra-identity-and-auth): Does adding a test with SRA way of overriding signer here makes sense? Probably useful as a
+    //  general test of SRAs support of provider your own signer, though not specific to the overriding of of AsyncAws4Signer.
+    //  Maybe still makes sense, by asserting that signAsync is called for streaming input operation (may belong to a different
+    //  test class)?
```

Comment:
+1 makes sense to add one. Here is probably as good a location as any.

Reply:
Here=codegen-generated-classes-test yes, I just thought here=AsyncSignerOverride not so, because of Async in the test class name. I'll rename the class to more general and add the test here.

Reply:
Well, can't add that test until another task is done - to allow putAuthScheme on client, as a way to provider your own signer. So leaving the TODO, and resolving this comment for now.